api/ngrok/v1alpha1/ngroktrafficpolicy_types.go

```diff
@@ -0,0 +1,73 @@
+/*
+MIT License
+
+Copyright (c) 2022 ngrok, Inc.
+
+Permission is hereby granted, free of charge, to any person obtaining a copy
+of this software and associated documentation files (the "Software"), to deal
+in the Software without restriction, including without limitation the rights
+to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+copies of the Software, and to permit persons to whom the Software is
+furnished to do so, subject to the following conditions:
+
+The above copyright notice and this permission notice shall be included in all
+copies or substantial portions of the Software.
+
+THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+SOFTWARE.
+*/
+
+package v1alpha1
+
+import (
+	"encoding/json"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// EDIT THIS FILE!  THIS IS SCAFFOLDING FOR YOU TO OWN!
+// NOTE: json tags are required.  Any new fields you add must have json tags for the fields to be serialized.
+
+// NgrokTrafficPolicySpec defines the desired state of NgrokTrafficPolicy
+type NgrokTrafficPolicySpec struct {
+	// INSERT ADDITIONAL SPEC FIELDS - desired state of cluster
+	// Important: Run "make" to regenerate code after modifying this file
+
+	Policy json.RawMessage `json:"policy,omitempty"`
+}
+
+// NgrokTrafficPolicyStatus defines the observed state of NgrokTrafficPolicy
+type NgrokTrafficPolicyStatus struct {
+	// INSERT ADDITIONAL STATUS FIELD - define observed state of cluster
+	// Important: Run "make" to regenerate code after modifying this file
```

Comment:
This status should also contain a policy string
```suggestion
		Policy json.RawMessage `json:"policy,omitempty"`
```

Reply:
Curious, why is the duplicated storage of the json needed?

Reply:
The `spec` describes the desired state while the `status` describes the current state, but I guess since we don't need to do any complex steps to reconcile the difference between the two we could only use the `spec` field and always construct on change.

Reply:
ahh, well if they're just conventionally mirroring like that I see why we're putting policy here also. Thanks for clarifying 🚀